### PR TITLE
Migrate term from a taxonomy to another one

### PR DESCRIPTION
**Pull request by Mte90**
The new command is avalaible!

I need only to do the unit tests but I will work on them but before to do that I want a feedback about the warning in the code.

Example output:
```
Post 218854 assigned to new term!
Post 218860 assigned to new term!
Post 218870 assigned to new term!
Post 218879 assigned to new term!
Post 218892 assigned to new term!
Post 218893 assigned to new term!
Post 218903 assigned to new term!
Post 218913 assigned to new term!
Post 218922 assigned to new term!
Post 218957 assigned to new term!
Post 218974 assigned to new term!
Post 218995 assigned to new term!
Post 219050 assigned to new term!
Post 219060 assigned to new term!
Term migrated!
Old Term removed!
Success: Migration of `9190` term for 5494 posts
```

### CODE REVIEW

**Comment by desrosj** [comment]
@Mte90 Are you able to write some tests for this?

**Comment by Mte90** [comment]
Yes I can do it but I am not sure about the warning (specifically about the text messages) printed in the console.

**Comment by schlessera** [comment]
Let me do a quick review of the text messages then so you now what final strings to test against...

File: src/Term_Command.php [hunk 1]
```diff
@@ -636,6 +636,95 @@ public function recount( $args ) {
...
+
+				WP_CLI::line( "Post {$post} assigned to new term!" );
+			}
+
+			clean_post_cache( $post );
+		}
+
+		clean_term_cache( $term->term_id );
+
+		WP_CLI::line( "Term migrated!" );
```
Comment (schlessera) [inline src/Term_Command.php]:
```suggestion
		WP_CLI::log( "Term '{$term->slug}' migrated!" );
```

File: src/Term_Command.php [hunk 2]
```diff
@@ -636,6 +636,95 @@ public function recount( $args ) {
...
+
+		$del = wp_delete_term( $term->term_id, $original_taxonomy->name );
+
+		if ( is_wp_error( $del ) ) {
+			WP_CLI::error( "On deleting the term, " . $del->get_error_message() );
+		}
+
+		WP_CLI::line( "Old Term removed!" );
+		WP_CLI::success( "Migration of `{$term_reference}` term for " . count( $posts ) . " posts" );
```
Comment (schlessera) [inline src/Term_Command.php]:
```suggestion
		$post_count  = count( $posts );
		$post_plural = WP_CLI\Utils\pluralize( 'post', $post_count )
		WP_CLI::success( "Migrated the term '{$term->slug}' from taxonomy '{original_taxonomy}' to taxonomy '{destination_taxonomy}' for {$post_count} {$post_plural}" );
```

File: src/Term_Command.php [hunk 3]
```diff
@@ -636,6 +636,95 @@ public function recount( $args ) {
...
+
+		WP_CLI::line( "Term migrated!" );
+
+		$del = wp_delete_term( $term->term_id, $original_taxonomy->name );
+
+		if ( is_wp_error( $del ) ) {
+			WP_CLI::error( "On deleting the term, " . $del->get_error_message() );
+		}
+
+		WP_CLI::line( "Old Term removed!" );
```
Comment (schlessera) [inline src/Term_Command.php]:
```suggestion
		WP_CLI::log( "Old instance of term '{$term->slug} removed from its original taxonomy." );
```

File: src/Term_Command.php [hunk 4]
```diff
@@ -636,6 +636,95 @@ public function recount( $args ) {
...
+	 * [--by=<field>]
+	 * : Explicitly handle the term value as a slug or id.
+	 * ---
+	 * default: id
+	 * options:
+	 *   - slug
+	 *   - id
+	 * ---
+	 *
+	 * [--from=<taxonomy>]
```
Comment (schlessera) [inline src/Term_Command.php]: This should not be optional, and we should require users to always explicitly specify the `--from` and `--to` taxonomies.

File: src/Term_Command.php [hunk 5]
```diff
@@ -636,6 +636,95 @@ public function recount( $args ) {
...
+	 * : Taxonomy slug of the term to migrate.
+	 * ---
+	 * default: category
+	 * ---
+	 *
+	 * [--to=<taxonomy>]
```
Comment (schlessera) [inline src/Term_Command.php]: This should not be optional either.

File: src/Term_Command.php [hunk 6]
```diff
@@ -636,6 +636,95 @@ public function recount( $args ) {
...
+	 * : Taxonomy slug to migrate.
+	 * ---
+	 * default: post_tag
+	 * ---
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     # Migrate a category's term (video) to tag taxonomy.
```
Comment (schlessera) [inline src/Term_Command.php]: Example will need to be updated.

**Comment by Mte90** [comment]
I will work on the other changes and tests in the next days.

**Comment by schlessera** [comment]
Thanks for the PR, @Mte90!

**Pull request merged**